app/src/main/kotlin/com/simplemobiletools/flashlight/activities/MainActivity.kt

```diff
@@ -1,332 +1,367 @@
-                mCameraImpl!!.onCameraNotAvailable()
-            }
-        })
-        if (config.turnFlashlightOn) {
-            mCameraImpl!!.enableFlashlight()
+        if (!resources.getBoolean(com.simplemobiletools.commons.R.bool.hide_google_relations)) {
+            faqItems.add(FAQItem(com.simplemobiletools.commons.R.string.faq_2_title_commons, com.simplemobiletools.commons.R.string.faq_2_text_commons))
+            faqItems.add(FAQItem(com.simplemobiletools.commons.R.string.faq_6_title_commons, com.simplemobiletools.commons.R.string.faq_6_text_commons))
         }
-        setupBrightness()
-    }
 
-    private fun setupStroboscope() {
-        binding.stroboscopeBar.apply {
-            max = (MAX_STROBO_DELAY - MIN_STROBO_DELAY).toInt()
-            progress = config.stroboscopeProgress
-            onSeekBarChangeListener { progress ->
-                val frequency = max - progress + MIN_STROBO_DELAY
-                mCameraImpl?.stroboFrequency = frequency
-                config.stroboscopeFrequency = frequency
-                config.stroboscopeProgress = progress
-            }
-        }
+        startAboutActivity(R.string.app_name, 0, BuildConfig.VERSION_NAME, faqItems, true)
     }
 
-    private fun toggleStroboscope(isSOS: Boolean) {
+    private fun toggleStroboscope(isSOS: Boolean, launcher: ManagedActivityResultLauncher<String, Boolean>) {
         // use the old Camera API for stroboscope, the new Camera Manager is way too slow
         if (isNougatPlus()) {
             cameraPermissionGranted(isSOS)
         } else {
-            handlePermission(PERMISSION_CAMERA) {
-                if (it) {
-                    cameraPermissionGranted(isSOS)
-                } else {
-                    toast(R.string.camera_permission)
-                }
+            val permission = Manifest.permission.CAMERA
+            if (ContextCompat.checkSelfPermission(this, permission) == PackageManager.PERMISSION_GRANTED) {
+                cameraPermissionGranted(isSOS)
+            } else {
+                launcher.launch(permission)
             }
         }
     }
 
-    private fun setupBrightness() {
-        binding.brightnessBar.apply {
-            max = mCameraImpl?.getMaximumBrightnessLevel() ?: MIN_BRIGHTNESS_LEVEL
-            progress = mCameraImpl?.getCurrentBrightnessLevel() ?: MIN_BRIGHTNESS_LEVEL
-            onSeekBarChangeListener { level ->
-                val newLevel = level.coerceAtLeast(MIN_BRIGHTNESS_LEVEL)
-                mCameraImpl?.updateBrightnessLevel(newLevel)
-                config.brightnessLevel = newLevel
-            }
+    private fun getPermissionResultHandler(isSos: Boolean): (Boolean) -> Unit = {
+        handlePermissionResult(isSos, it)
+    }
+
+    private fun handlePermissionResult(isSos: Boolean, granted: Boolean) {
+        if (granted) {
+            cameraPermissionGranted(isSos)
+        } else {
+            toast(R.string.camera_permission)
```

Comment:
Do we even need this function at all?

Reply:
maybe copy the contents of this function to the one that calls it...

Reply:
Yeah, my bad, not needed at all, I will move it to its call site.